okhttp/src/test/java/io/grpc/okhttp/OkHttpTlsUpgraderTest.java

```diff
@@ -32,4 +34,44 @@
             || OkHttpTlsUpgrader.TLS_PROTOCOLS.indexOf(Protocol.GRPC_EXP)
                 < OkHttpTlsUpgrader.TLS_PROTOCOLS.indexOf(Protocol.HTTP_2));
   }
+
+  @Test public void upgrade_canonicalizeHost() {
```

Comment:
Tests coverage look good to me (with the addition of a case to ensure we don't strip out additional brackets for malformed URIs). Out of curiosity, what methodology went into choosing the test hosts?

Reply:
Will update to ensure we don't strip out additional brackets. 

I lifted these from okhttp's [HostnameVerifierTest](https://github.com/square/okhttp/blob/master/okhttp-tests/src/test/java/okhttp3/internal/tls/HostnameVerifierTest.java#L519), so I'll add some attribution.

Reply:
Let's simplify the tests somewhat here, which will also remove the need for attribution - we actually don't care about all of the edge cases that the linked test case covers, because we don't need to distinguish between IP addresses and hostnames. Probably just the following are enough:

```
assertEquals("::1", canonicalizeHost("::1"));
assertEquals("::1", canonicalizeHost("[::1]"));
assertEquals("127.0.0.1", canonicalizeHost("127.0.0.1"));
assertEquals("some.long.url.com", canonicalizeHost("some.long.url.com"));

// Extra square brackets in a malformed URI are retained
assertEquals("[::1]", canonicalizeHost("[[::1]]"));
```

Also, does it make sense to rename the test method to just `canonicalizeHost`, as this isn't testing any other part of the upgrade logic?

Reply:
👍 Is `canonicalizeHosts` ok to avoid clash with the static import of `canonicalizeHost`?

Reply:
sounds good to me